validator/client/src/main/java/tech/pegasys/teku/validator/client/RuntimeProposerConfig.java

```diff
@@ -106,28 +125,44 @@ private synchronized void read(final Path path) {
 
   static class Config {
     private final Eth1Address feeRecipient;
+    private final UInt64 gasLimit;
+
+    public Config(final Optional<Eth1Address> feeRecipient, final Optional<UInt64> gasLimit) {
+      this.feeRecipient = feeRecipient.orElse(null);
+      this.gasLimit = gasLimit.orElse(null);
+    }
 
-    public Config(final Eth1Address feeRecipient) {
-      this.feeRecipient = feeRecipient;
+    public Optional<Eth1Address> getFeeRecipient() {
+      return this.feeRecipient == null ? Optional.empty() : Optional.of(this.feeRecipient);
     }
 
-    public Eth1Address getFeeRecipient() {
-      return feeRecipient;
+    public Optional<UInt64> getGasLimit() {
+      return this.gasLimit == null ? Optional.empty() : Optional.of(this.gasLimit);
     }
   }
 
   static class ConfigBuilder {
     private Eth1Address feeRecipient;
+    private UInt64 gasLimit;
 
     public ConfigBuilder() {}
 
-    public ConfigBuilder feeRecipient(final Eth1Address feeRecipient) {
-      this.feeRecipient = feeRecipient;
+    public ConfigBuilder feeRecipient(final Optional<Eth1Address> feeRecipient) {
+      this.feeRecipient = feeRecipient.orElse(null);
+      return this;
+    }
+
+    public ConfigBuilder gasLimit(final Optional<UInt64> gasLimit) {
+      this.gasLimit = gasLimit.orElse(null);
       return this;
     }
 
     public Config build() {
-      return new Config(feeRecipient);
+      Optional<Eth1Address> feeRecipient =
```

Comment:
once the above are optional these reduce a lot, but otherwise `Optional.ofNullable`